_posts/2017-11-02-totw-107.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Tip of the Week #107: Reference Lifetime Extension"
+layout: tips
+sidenav: side-nav-tips.html
+published: draft
+permalink: tips/107
+type: markdown
+---
+
+# Tip of the Week #107: Reference Lifetime Extension
+
+Originally posted as totw/107 on 2015-12-10
+
+By Titus Winters (titus@google.com)
+
+There have been some reports of confusion about references and lifetimes after
+abseil.io/tips/101, so in this Tip we'll dive more into the question, "When does
+reference lifetime extension apply?"
+
+```c++
+string Foo::GetName();
+
+const string& name = obj.GetName();  // Is this safe/legal?
+```
+
+In short lifetimes of temporaries (referents) will be extended *if and only if*
+
+*   A local `const T&` (or `T&&`, although Google style generally ignores that)
+    is initialized to the result of an expression (usually a function call)
+    returning a temporary `T` *or* the `T` subobject of a temporary (e.g. a
+    struct containing `T`).
```

Comment:
. ->, **except** for the following caveats:

(I feel you need to introduce the exceptions below.)

In that case, you can remove the 

"This doesn't work", "The case where" in the following

Reply:
None of these are caveats, they are clarifications.  The rule as stated is perfectly correct, but obscure, hence the clarifications.

Reply:
Then I'd introduce them as clarifications. It isn't entirely clear from the context.

As well, that means we have only one bullet (the main one) and a bunch of sub-bullets, which I think is confusing. Basic rule is that you don't need a list if you have 1 item :) 

I'd merge the original bullet into the intro sentence and note the clarifications as a separate set of bullets. I.e.

In short, lifetimes of temporaries (referents) will be extended *if and only if* a local `const T&` (or `T&&`, although Google style generally ignores that) is initialized to the result of an expression (usually a function call) returning a temporary `T` *or* the `T` sub-object of a temporary (e.g. a  struct containing `T`). As well, note the following clarifications:

* This doesn't work ...

Reply:
Done